src/openzeppelin/account/library.cairo

```diff
@@ -1,334 +1,250 @@
+        new_public_key: felt
+    ) {
+        assert_only_self();
+        Account_public_key.write(new_public_key);
+        return ();
+    }
+
+    //
+    // Business logic
+    //
 
     func is_valid_signature{
-            syscall_ptr : felt*,
-            pedersen_ptr : HashBuiltin*,
-            range_check_ptr,
-            ecdsa_ptr: SignatureBuiltin*
-        }(
-            hash: felt,
-            signature_len: felt,
-            signature: felt*
-        ) -> (is_valid: felt):
-        let (_public_key) = Account_public_key.read()
-
-        # This interface expects a signature pointer and length to make
-        # no assumption about signature validation schemes.
-        # But this implementation does, and it expects a (sig_r, sig_s) pair.
-        let sig_r = signature[0]
-        let sig_s = signature[1]
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        ecdsa_ptr: SignatureBuiltin*,
+    }(hash: felt, signature_len: felt, signature: felt*) -> (is_valid: felt) {
+        let (_public_key) = Account_public_key.read();
+
+        // This interface expects a signature pointer and length to make
+        // no assumption about signature validation schemes.
+        // But this implementation does, and it expects a (sig_r, sig_s) pair.
+        let sig_r = signature[0];
+        let sig_s = signature[1];
 
         verify_ecdsa_signature(
-            message=hash,
-            public_key=_public_key,
-            signature_r=sig_r,
-            signature_s=sig_s)
+            message=hash, public_key=_public_key, signature_r=sig_r, signature_s=sig_s
+        );
 
-        return (is_valid=TRUE)
-    end
+        return (is_valid=TRUE);
+    }
 
     func is_valid_eth_signature{
-            syscall_ptr : felt*,
-            pedersen_ptr : HashBuiltin*,
-            bitwise_ptr: BitwiseBuiltin*,
-            range_check_ptr
-        }(
-            hash: felt,
-            signature_len: felt,
-            signature: felt*
-        ) -> (is_valid: felt):
-        alloc_locals
-        let (_public_key) = get_public_key()
-        let (__fp__, _) = get_fp_and_pc()
-
-        # This interface expects a signature pointer and length to make
-        # no assumption about signature validation schemes.
-        # But this implementation does, and it expects a the sig_v, sig_r,
-        # sig_s, and hash elements.
-        let sig_v : felt = signature[0]
-        let sig_r : Uint256 = Uint256(low=signature[1], high=signature[2])
-        let sig_s : Uint256 = Uint256(low=signature[3], high=signature[4])
-        let (high, low) = split_felt(hash)
-        let msg_hash : Uint256 = Uint256(low=low, high=high)
-
-        let (local keccak_ptr : felt*) = alloc()
-
-        with keccak_ptr:
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
```

Comment:
why flipping these two? or was it the formatter?

Reply:
For consistency with the implicit arg order in `is_valid_signature`. We can change it back if preferred

Reply:
the criteria so far was explicitly typed params first, i'd apply that wherever needed